odoo/tools/mail.py

```diff
@@ -426,6 +426,20 @@ def append_content_to_html(html, content, plaintext=True, preserve=False, contai
         return '%s%s' % (html, content)
     return '%s%s%s' % (html[:insert_location], content, html[insert_location:])
 
+
+def prepend_html_content(html_body, html_content):
+    """Prepend some HTML content at the beginning of an other HTML content."""
+    html_content = re.sub(r'(?i)(</?(?:html|body|head|!\s*DOCTYPE)[^>]*>)', '', html_content)
```

Comment:
removing via regex is kind of hard, for instance, you match `<body>` but not `< body>`. Wouldn't it be better to use lxml or other specific parsing tool?

Reply:
`< body>` is not a valid HTML tag

And also, just parsing the HTML will be a lot slower than just applying the regex
(See: https://pastebin.com/hwsvQ6AZ )

![_tmp_screenshot](https://user-images.githubusercontent.com/54316226/83146538-c2462f80-a0f6-11ea-9f72-771c2af45e69.png)

I think it depends if we want a super robust solution or a fast solution

Reply:
@mart-e